File: frameworks/Rust/rocket/src/schema.rs

table! {
    product (id) {
        id -> Int4,
        title -> Nullable<Varchar>,
        body_sanitized -> Nullable<Varchar>,
    }
}

table! {
    retailer (id) {
        id -> Int4,
        GSTIN -> Nullable<Varchar>,
        Business_name -> Nullable<Varchar>,
        Contact_person -> Nullable<Varchar>,
        Contact_number -> Nullable<Int4>,
        Contact_address -> Nullable<Varchar>,
        Contact_emailId -> Nullable<Varchar>,
        Status -> Nullable<Varchar>,
        Outlet_limit -> Nullable<Int4>,
    }
}

table! {
    store (id) {
        id -> Int4,
        name -> Nullable<Varchar>,
    }
}

allow_tables_to_appear_in_same_query!(
    product,
    retailer,
    store,
);

File: bench_pg/JS/elysia-drizzle/src/schema.ts
import { InferModel } from "drizzle-orm";
import { pgTable, serial, text, varchar, integer } from "drizzle-orm/pg-core";

export const retailers = pgTable("retailer", {
  id: serial("id").primaryKey(),
  GSTIN: varchar("GSTIN", { length: 15 }),
  businessName: varchar("Business_name", { length: 100 }),
  contactPerson: varchar("Contact_person", { length: 50 }),
  contactNumber: integer("Contact_number"),
  contactAddress: varchar("Contact_address", { length: 100 }),
  contactEmail: varchar("Contact_emailId", { length: 50 }),
  status: varchar("Status", { length: 10 }),
  outletLimit: integer("Outlet_limit"),
});

export type RetailerType = InferModel<typeof retailers>;
